src/app/shared/services/api/search.repo.ts

```diff
@@ -77,10 +77,13 @@ export class SearchRepo extends BaseRepo {
       query,
       archiveId,
       publicOnly: true,
-      numberOfResults: limit,
       ...tagsDict,
     };
 
+    if (limit) {
+      data['numberOfResults'] = limit;
```

Comment:
We can and probably should avoid accessing this property with a string by using an explicit type definition for `data` instead.